pkg/controller/util/endpoint/controller_utils_test.go

```diff
@@ -99,9 +99,8 @@ func TestDetermineNeededServiceUpdates(t *testing.T) {
 	}
 }
 
-// There are 3*5 possibilities(3 types of RestartPolicy by 5 types of PodPhase).
-// Not listing them all here. Just listing all of the 3 false cases and 3 of the
-// 12 true cases.
+// Only Pods that are Ready or Unready (if the Service tolerate unready )
+// should be present on Endpoints or EndpointSlices, terminal pods should not.
```

Comment:
I think this comment is more confusing than it is helpful. The Ready or Unready status of a Pod is somewhat orthogonal to whether it's terminal. I think there have been previous bugs where ready==true while terminal was also true. Maybe just remove this comment?

Reply:
removed